pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -246,8 +246,8 @@ public void removeAgedDeletedSegments(int retentionInDays) {
 
         if (numFilesDeleted == targetFiles.length) {
           // Delete directory if it's empty
-          if (!pinotFS.delete(currentURI)) {
-            LOGGER.warn("The directory {} cannot be removed.", currentDir);
+          if (!pinotFS.delete(uri, true)) {
```

Comment:
should be 'false', in case a segment was deleted, and was moved into the deleted dir for that table _after_ we got the segment list already in the deleted dir.